packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -48,6 +51,22 @@ abstract class GestureRecognizer extends GestureArenaMember {
   /// Returns a very short pretty description of the gesture that the
   /// recognizer looks for, like 'tap' or 'horizontal drag'.
   String toStringShort() => toString();
+
+  /// Invoke a callback provided by the application and log any exceptions.
+  dynamic invokeCallback(RecognizerCallback callback) {
+    dynamic result;
+    try {
+      result = callback();
+    } catch (exception, stack) {
+      FlutterError.reportError(new FlutterErrorDetails(
+        exception: exception,
+        stack: stack,
+        library: 'recognizer',
+        context: 'while handling a gesture',
+      ));
```

Comment:
Ideally you could add an information collector that includes:
* `'$this'`
* the name of the callback (passed in as an argument to this method)